client/blocks/reader-suggested-follows/dialog.jsx

```diff
@@ -1,71 +1,119 @@
 import { Dialog } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
 import { useEffect } from 'react';
-import SuggestedFollowItem from 'calypso/blocks/reader-suggested-follows';
-import { useRelatedSites } from 'calypso/data/reader/use-related-sites';
+import { SuggestedFollowItem } from 'calypso/blocks/reader-suggested-follows';
 import { READER_SUGGESTED_FOLLOWS_DIALOG } from 'calypso/reader/follow-sources';
 import { useDispatch } from 'calypso/state';
 import { recordReaderTracksEvent } from 'calypso/state/reader/analytics/actions';
-
+import { useRecommendOrRelatedSitesQuery } from './hooks/use-recommend-or-related-sites-query';
+import { RecommendedFeed } from './recommended-feed';
 import './style.scss';
 
-const ReaderSuggestedFollowsDialog = ( { onClose, siteId, postId, isVisible } ) => {
+const ReaderSuggestedFollowsDialog = ( { onClose, siteId, postId, isVisible, author = {} } ) => {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const { data, isLoading } = useRelatedSites( siteId, postId );
+	const userName = author?.name;
+
+	const { data, isLoading, resourceType } = useRecommendOrRelatedSitesQuery(
+		{
+			author,
+			siteId,
+			postId,
+		},
+		{
+			enabled: isVisible,
+		}
+	);
 
 	useEffect( () => {
-		if ( isVisible ) {
-			dispatch( recordReaderTracksEvent( 'calypso_reader_suggested_follows_dialog_viewed' ) );
+		if ( isVisible && resourceType ) {
+			dispatch(
+				recordReaderTracksEvent( 'calypso_reader_suggested_follows_dialog_viewed', {
+					resource_type: resourceType,
+				} )
+			);
 		}
-	}, [ isVisible, dispatch ] );
+	}, [ isVisible, dispatch, resourceType ] );
 
 	// If we are no longer loading and no data available, don't show the dialog
-	if ( ! isLoading && data === undefined ) {
+	if ( ! isLoading && resourceType === null ) {
 		return null;
```

Comment:
Genuine question here, if we return `null` without handling the `onClose` could this cause any problems? I ask because there is a state that is managed in `ReaderFeedHeaderFollow` that is true when opened but is not set to false if we bail here. I know you didn't add this part, but was curious.

Reply:
I updated the code to remove this logic:

https://github.com/Automattic/wp-calypso/pull/104927/commits/411cf8a5cb8c964c232a179ab0addec59c7e2fa8#diff-0f63a79dbba7afcb6d94b00c047aac192cc33f1d891d7f4931419eef7fac3419R80-R85

I renamed the isSuccess flag to isFetched to represent the intent better. 

But let's rethink our strategy for showing it in the future. 
I ended up adding tracking because I want to know if we should have an empty state or use any other strategy. |